app/models/user.rb

```diff
@@ -0,0 +1,43 @@
+class User < ApplicationRecord
+  # Remember to create a migration!
+    include BCrypt
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :username, presence: true
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true
+
+  has_many :tweets
+  #people that I am following
+  has_many :follows, foreign_key: :follower_id
+  has_many :followings, through: :follows, source: :follower
+  #people that are following me
+  # **HAVE TO CHANGE SAME NAME TO AVOID OVERWRITE**
+  has_many :follows, foreign_key: :followed_id
```

Comment:
why would you need to repeat any of this? Wouldn't it just be followers and followed?